crates/nu-protocol/src/debugger/profiler.rs

```diff
@@ -262,6 +267,23 @@ fn expr_to_string(engine_state: &EngineState, expr: &Expr) -> String {
     }
 }
 
+// Find a file name and a line number (indexed from 1) of a span
+fn find_file_of_span(engine_state: &EngineState, span: Span) -> Option<(&str, usize)> {
+    for file in engine_state.files() {
+        if file.covered_span.start < span.start && file.covered_span.end > span.start {
+            // count the number of lines between file start and the searched span start
+            let chunk =
+                engine_state.get_span_contents(Span::new(file.covered_span.start, span.start));
+            let nlines = chunk.lines().count();
+            let line_num = if nlines <= 1 { 1 } else { nlines + 1 };
```

Comment:
Not sure if this is correct but ok.

Reply:
I noticed that with only one line, the `nlines + 1` would incorrectly give 2 as the line number, so I special-cased it. This seems to work correctly on regular inputs and files, but I didn't test it thoroughly. There might be some edge cases which we can patch as we discover them.

Reply:
I think this should work? Let me know if anything looks wrong.
```suggestion
            // account for leading part of current line being counted as a separate line
            let line_num = if chunk.last() == Some(&b'\n') {
                nlines + 1
            } else {
                nlines
            };

            // first line has no previous line, clamp up to `1`
            let line_num = usize::max(line_num, 1);
```